var/spack/repos/builtin/packages/python/package.py

```diff
@@ -1122,82 +853,317 @@ def setup_dependent_build_environment(self, env, dependent_spec):
+                self.pkg.command("-c", "import crypt")
+
+
+class AutotoolsBuilder(autotools.AutotoolsBuilder, RunAfter, BuildEnvironment):
+    # An in-source build with --enable-optimizations fails for python@3.X
+    build_directory = "spack-build"
+
+    def configure_args(self):
+        spec = self.spec
+        config_args = []
+        cflags = []
+
+        # setup.py needs to be able to read the CPPFLAGS and LDFLAGS
+        # as it scans for the library and headers to build
+        link_deps = spec.dependencies(deptype="link")
+
+        if link_deps:
+            # Header files are often included assuming they reside in a
+            # subdirectory of prefix.include, e.g. #include <openssl/ssl.h>,
+            # which is why we don't use HeaderList here. The header files of
+            # libffi reside in prefix.lib but the configure script of Python
+            # finds them using pkg-config.
+            cppflags = " ".join("-I" + spec[dep.name].prefix.include for dep in link_deps)
+
+            # Currently, the only way to get SpecBuildInterface wrappers of the
+            # dependencies (which we need to get their 'libs') is to get them
+            # using spec.__getitem__.
+            ldflags = " ".join(spec[dep.name].libs.search_flags for dep in link_deps)
+
+            config_args.extend(["CPPFLAGS=" + cppflags, "LDFLAGS=" + ldflags])
+
+        if "+optimizations" in spec:
+            config_args.append("--enable-optimizations")
+            # Prefer thin LTO for faster compilation times.
+            if "@3.11.0: %clang@3.9:" in spec or "@3.11.0: %apple-clang@8:" in spec:
+                config_args.append("--with-lto=thin")
             else:
-                os.remove(dst)
+                config_args.append("--with-lto")
+            config_args.append("--with-computed-gotos")
 
-    def test(self):
-        # do not use self.command because we are also testing the run env
-        exe = self.spec["python"].command.name
+        if spec.satisfies("@3.7 %intel"):
+            config_args.append("--with-icc={0}".format(spack_cc))
 
-        # test hello world
-        msg = "hello world!"
-        reason = "test: running {0}".format(msg)
-        options = ["-c", 'print("{0}")'.format(msg)]
-        self.run_test(exe, options=options, expected=[msg], installed=True, purpose=reason)
+        if "+debug" in spec:
+            config_args.append("--with-pydebug")
+        else:
+            config_args.append("--without-pydebug")
 
-        # checks import works and executable comes from the spec prefix
-        reason = "test: checking import and executable"
-        options = ["-c", "import sys; print(sys.executable)"]
-        self.run_test(
-            exe, options=options, expected=[self.spec.prefix], installed=True, purpose=reason
-        )
+        if "+shared" in spec:
+            config_args.append("--enable-shared")
+        else:
+            config_args.append("--disable-shared")
```

Comment:
Here you can:
```suggestion
        config_args.extend(self.enable_or_disable("shared"))
```
and likewise for the other options below, where it makes sense.